components/style/stylist.rs

```diff
@@ -1628,6 +1671,33 @@ impl<'a> Iterator for CascadeDataIter<'a> {
     }
 }
 
+/// Iterator over `PerOriginExtraStyleData`, from highest level (user) to lowest
+/// (user agent).
+///
+/// We rely on this specific order for correctly looking up the @font-face
+/// and @counter-style rules.
+#[cfg(feature = "gecko")]
+pub struct ExtraStyleDataIter<'a> {
+    extra_style_data: &'a ExtraStyleData,
+    cur: usize,
+}
+
+#[cfg(feature = "gecko")]
+impl<'a> Iterator for ExtraStyleDataIter<'a> {
```

Comment:
I suspect it's going to be nicer to have a `PerOrigin<T>` that abstracts all this stuff.